batch_job/src/batch_job_amazon_batch.c

```diff
@@ -0,0 +1,567 @@
+}
+
+static char* generate_s3_cp_cmds(char* files, char* src, char* dst){
+	char* env_var = initialized_data.master_env_prefix;
+	int i;
+	int files_split_num = 0;
+	char** files_split;
+	split_comma_list(files, &files_split_num, &files_split);
+
+	char* new_cmd=malloc(sizeof(char)*1);
+        new_cmd[0]='\0';
+	if(files_split_num > 0){
+		char* copy_cmd_prefix = string_format("%s aws s3 cp ", env_var);
+		for(i=0; i<files_split_num; i++){
+			char* tmp = string_format("%s %s/%s %s",copy_cmd_prefix, src, files_split[i], dst);
+			char* tmp2 = string_format("%s\n%s\n",new_cmd,tmp);
+			free(new_cmd);
+			free(tmp);
+			new_cmd = tmp2;
+		}
+	}
+	clean_str_array(files_split_num,&files_split);
+	return new_cmd;
+}
+
+static char* chmod_all(char* files){
+	int i;
+	int files_split_num = 0;
+	char** files_split;
+	split_comma_list(files,&files_split_num,&files_split);
+	char* new_cmd=malloc(sizeof(char)*1);
+	new_cmd[0]='\0';
+	if(files_split_num > 0){
+		for(i=0; i<files_split_num; i++){
+			char* tmp = string_format("chmod +x %s",files_split[i]);
+			char* tmp2 = string_format("%s\n%s",new_cmd,tmp);
+			free(new_cmd);
+			free(tmp);
+			new_cmd=tmp2;
+		}
+	}
+	clean_str_array(files_split_num,&files_split);
+	return new_cmd;
+}
+
+static void upload_cmd_file(char* bucket_name, char* input_files, char* output_files, char* cmd, unsigned int jobid){
+	char* env_var = initialized_data.master_env_prefix;
+	//Create command to pull files from s3 and into local space to work on
+	char* bucket = string_format("s3://%s",bucket_name);
+	char* cpy_in = generate_s3_cp_cmds(input_files,bucket,"./");
+	char* chmod = chmod_all(input_files);
+	//run the actual command
+	char* cmd_tmp = string_format("%s\n%s\n%s\n",cpy_in,chmod,cmd);
+	free(cpy_in);
+	//copy out any external files
+	char* cpy_out = generate_s3_cp_cmds(output_files,"./",bucket);
+	cmd_tmp = string_format("%s\n%s\n",cmd_tmp,cpy_out);
+
+	//add headder
+	char* final_cmd = string_format("#!/bin/sh\n%s",cmd_tmp);
+	free(cmd_tmp);
+
+
+	//write out to file	
+	FILE* tmpfile = fopen("TEMPFILE.sh","w+");
+	fwrite(final_cmd,sizeof(char),strlen(final_cmd),tmpfile);
+	fclose(tmpfile);
+	free(final_cmd);
+	
+	//make executable and put into s3
+	sh_system("chmod +x TEMPFILE.sh");
+	cmd_tmp = string_format("%s /usr/bin/time -f \"Submitting cmd_file Job %s_%u: %%e\" -a -o 'BatchJobAmazonBatchTimings.txt' aws s3 cp ./TEMPFILE.sh s3://%s/COMAND_FILE_%u.sh",env_var,queue_name,jobid,bucket_name,jobid);
+	sh_system(cmd_tmp);
+	free(cmd_tmp);
+	remove("TEMPFILE.sh");	
+
+}
+
+static char* aws_submit_job(char* job_name, char* properties_string){
+	char* queue = queue_name;
+	char* env_var = initialized_data.master_env_prefix;
+	//submit the job-def
+	char* tmp = string_format("%s /usr/bin/time -f \"Registering Job %s: %%e\" -a -o 'BatchJobAmazonBatchTimings.txt' aws batch register-job-definition --job-definition-name %s_def --type container --container-properties \"%s\"",env_var,job_name,job_name, properties_string);
```

Comment:
why /usr/bin/time?

Reply:
needed to ensure specific form of time so it will accept that format string.